### PR TITLE
Add docs for Kargo installer for Kubernetes

### PR DESCRIPTION
**Pull request by bogdando**
Add documentation for Getting Started guides.
Kargo is an incubated Kubernetes project
https://github.com/kubernetes-incubator/kargo.

Closes https://github.com/kubernetes-incubator/kargo/issues/622

Signed-off-by: Bogdan Dobrelya <bogdando@mail.ru>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2183)

### CODE REVIEW

**Comment by bogdando** [comment]
PTAL
@mikedanese (champion)
@bgrant0607 (sponsor)

File: docs/getting-started-guides/kargo.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+* Enusre the netchecker-agent's pods can resolve DNS requests and ping each over within the default namespace.
+  Those pods mimic similar behavior of the rest of the workloads and serve as cluster health indicators.
+
+## Explore contributed add-ons
+
+See the [list of contributed playbooks](https://github.com/kubernetes-incubator/kargo/tree/master/contrib)
+to explore other deployment options.
+
+## What's next
+
```
Comment (mattymo) [inline docs/getting-started-guides/kargo.md]: This shouldn't be empty. Point to roadmap file here

File: docs/getting-started-guides/kargo.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+## Cleanup
+
+To delete your toy cluster you can apply the
```
Comment (mattymo) [inline docs/getting-started-guides/kargo.md]: toy cluster?

File: docs/getting-started-guides/kargo.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  * [Versions](https://github.com/kubernetes-incubator/kargo#versions-of-supported-components)
+    of components.
+  * Additional node groups like [bastion hosts](https://github.com/kubernetes-incubator/kargo/blob/master/docs/ansible.md#bastion-host) or
+    [Calico BGP route reflectors](https://github.com/kubernetes-incubator/kargo/blob/master/docs/calico.md#optional--bgp-peering-with-border-routers).
+* Plan custom deployment steps, if any, or use the default composition layer in the
+  [cluster definition file](https://github.com/kubernetes-incubator/kargo/blob/master/cluster.yml).
+  Taking the best from Ansible world, Kargo allows users to execute arbitrary steps via the
+  ``ansible-playbook`` with given inventory, playbooks, data overrides and tags, limits, batches
+  of nodes to deploy and so on.
+* For large deployments, like hundreds of nodes,
```
Comment (mattymo) [inline docs/getting-started-guides/kargo.md]: For large deployments (100+ nodes), you may want to [tweak things] for best results.

File: docs/getting-started-guides/kargo.md (hunk 2 above)
Comment (bogdando) [inline docs/getting-started-guides/kargo.md]: s/toy/production ? :)

Comment (mattymo) [inline docs/getting-started-guides/kargo.md]: Other choices: test, demo, lab, local, scratch

File: docs/getting-started-guides/kargo.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+to [kubeadm](../kubeadm) and [kops](../kops).
+
+## Creating a cluster
+
+### (1/4) Ensure the underlay [requirements](https://github.com/kubernetes-incubator/kargo#requirements) are met
+
+#### Checklist
+
+* You must have cloud instances or baremetal nodes running for your future Kubernetes cluster.
+  The easiest way to achieve that is to use the [kargo-cli tool](https://github.com/kubernetes-incubator/kargo/blob/master/docs/getting-started.md).
```
Comment (bogdando) [inline docs/getting-started-guides/kargo.md]: as @ant31 suggested, we should move the "easiest way" a step down

File: docs/getting-started-guides/kargo.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+* For large deployments (100+ nodes), you may want to
+  [tweak things](https://github.com/kubernetes-incubator/kargo/blob/master/docs/large-deployments.md)
+  for best results.
+
+### (3/4) Run the deployment
+
+#### Checklist
+
+* Apply deployment with [kargo-cli tool](https://github.com/kubernetes-incubator/kargo/blob/master/docs/getting-started.md)
+  or ``ansible-playbook`` manual commands.
```
Comment (bogdando) [inline docs/getting-started-guides/kargo.md]: (@ant31) we should add example commands like
```
cp -r inventory my_inventory
mv my_inventory/hosts.cfg.exampl my/inventory/hosts.cfg
# edit my_inventory/groups_vars/*.yaml
ansible-playbook -i my_inventory/hosts.cfg cluster.yaml -vv --private-key=~/.ssh/private_key
```

File: docs/getting-started-guides/kargo.md (hunk 4 above)
Comment (ant31) [inline docs/getting-started-guides/kargo.md]: 'The easiest way' -> 'a way'

File: docs/getting-started-guides/kargo.md (hunk 1 above)
Comment (bogdando) [inline docs/getting-started-guides/kargo.md]: done

File: docs/getting-started-guides/kargo.md (hunk 5 above)
Comment (bogdando) [inline docs/getting-started-guides/kargo.md]: I'll put those to the related Getting Started guide in Kargo, that is referenced here in a few places

Comment (bogdando) [inline docs/getting-started-guides/kargo.md]: done

**Comment by bogdando** [comment]
it's ready for review

**Comment by bogdando** [comment]
Folks, any chances to get that accepted?

**Comment by bogdando** [comment]
@mikedanese (champion)
@bgrant0607 (sponsor)
please review this patch, it is heavily required for the Kargo incubated project and users' UX

**Comment by idvoretskyi** [comment]
сс @jaredbhatti @kubernetes/sig-docs-misc

**Comment by bogdando** [comment]
Travis gate failure looks not related, any ideas why it fails? A master regression?

**Comment by janetkuo** [comment]
Travis failure is unrelated, see: #2210

**Comment by bogdando** [comment]
Travis recheck

**Comment by bogdando** [comment]
Sup folks, any chances to get this merged?

**Comment by chenopis** [comment]
@bogdando Sorry for the delay. I think the link [tool](https://github.com/kubernetes-incubator/kargo/blob/master/contrib/inventory_generator/inventory_generator.py) is broken. Otherwise it looks good to me.

**Comment by chenopis** [comment]
@bogdando I don't want to hold this up, so can you fix that link in a separate PR? Thx

**Pull request merged**